lib/c/arm8_intrinsics.h

```diff
@@ -0,0 +1,16 @@
+
+static inline uint64_t
+uint128_t_add_carry_u64(uint64_t cin, uint64_t x, uint64_t y, uint64_t* r) {
+  __uint128_t res = (__uint128_t)x + cin + y;
```

Comment:
I guess here you're relying on the fact that + is left-associative to get correct integer promotion rules?

Reply:
Yes. Is that not a good assumption?

Reply:
No that's fine, just checking how that's supposed to work, that stuff is notoriously tricky.